Ports/.port_include.sh

```diff
@@ -407,10 +407,10 @@ func_defined patch_internal || patch_internal() {
     # patch if it was not yet patched (applying patches multiple times doesn't work!)
     if [ -z "${IN_SERENITY_PORT_DEV:-}" ] && [ -d patches ]; then
         for filepath in patches/*.patch; do
-            filename=$(basename $filepath)
-            if [ ! -f "$workdir"/.${filename}_applied ]; then
-                run patch -p"$patchlevel" < "$filepath"
-                run touch .${filename}_applied
+            filename="$(basename \"${filepath}\")"
```

Comment:
Does `\"` work as expected here? As far as I know, one usually just puts normal `"` in the inner command.

Reply:
It's an escaped character. The reason I added them here is that `filepath` could have a space in the path, so it needs to be in double quotes to get passed to `basename` as a single argument.

If we really dislike the escaped double quotes we can instead do
```bash
filename="$(basename "'"'"${filepath}"'"'")"
```
or ignore this issue.

Reply:
In that case, the string indeed ends and `$(name}` is not in quotes.

However, for the case here, the inner command is wrapped into `$()`, so Bash knows that the string can't end yet, and adds it to the list for the inner command instead. Notably, adding escapes for the quotes breaks that behavior completely:

![Bildschirmfoto_2022-06-10_09-22-50](https://user-images.githubusercontent.com/16820960/173012536-b815ab80-af24-490b-95d4-a0fe43ac03aa.png)